pkg/engine/internal/executor/column.go

```diff
@@ -51,6 +51,18 @@ func NewScalar(value types.Literal, rows int) arrow.Array {
 		for range rows {
 			builder.Append(arrow.Timestamp(value))
 		}
+	case *array.ListBuilder:
+		//TODO(twhitney): currently only supporting string list, but we can add more types here as we need them
+		value, ok := value.Any().([]string)
+		if ok {
```

Comment:
nit: I personally prefer panics over silently ignoring programming errors. They make them easier to spot if our assumptions about the program are wrong.